### PR TITLE
feat: Upgrade to Webpack 4

### PR DESCRIPTION
**Pull request by erunion**
This upgrades all packages to Webpack 4 and removes a handful of packages that are not being used.

#### Webpack 3.10.0
```
Time: 31348ms
                            Asset     Size  Chunks                    Chunk Names
         ./example/demo-bundle.js  6.59 MB       0  [emitted]  [big]  demo
    ./example/reference-bundle.js  6.39 MB       1  [emitted]  [big]  reference
     ./example/demo-bundle.js.map  7.12 MB       0  [emitted]         demo
./example/reference-bundle.js.map  6.92 MB       1  [emitted]         reference
```

```
$ du -hs packages/*/dist
1.5M	packages/api-explorer/dist
156K	packages/api-logs/dist
```

#### Webpack 4.39.1
```
Time: 29296ms
                  Asset      Size     Chunks             Chunk Names
         demo-bundle.js  6.56 MiB       demo  [emitted]  demo
     demo-bundle.js.map  6.37 MiB       demo  [emitted]  demo
    reference-bundle.js  6.39 MiB  reference  [emitted]  reference
reference-bundle.js.map   6.2 MiB  reference  [emitted]  reference
```

```
$ du -hs packages/*/dist
1.5M	packages/api-explorer/dist
160K	packages/api-logs/dist
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,17 +39,15 @@
...
-    "style-loader": "^0.23.1",
-    "uglify-es": "^3.2.1",
-    "uglifyjs-webpack-plugin": "^1.2.3",
-    "webpack": "^3.10.0",
-    "webpack-dev-server": "^2.9.7",
-    "webpack-merge": "^4.1.1"
+    "style-loader": "^1.0.0",
+    "uglifyjs-webpack-plugin": "^2.2.0",
+    "webpack": "^4.39.1",
+    "webpack-cli": "^3.3.6",
```
[the remark this answers is not in the document]
Comment (erunion) [reply package.json]: Yeah, it's weird.

**Comment by erunion** [comment]
@domharrington Updated the PR description with dev bundle and dist package sizes. They're about still the same.

**Comment by domharrington** [comment]
I really do wanna reduce the bundle size. I've tried before to look into https://github.com/webpack-contrib/webpack-bundle-analyzer but it's always our dependency's dependencies that make up the bulk of it.

**Comment by erunion** [comment]
Running it I also see the same. RJSF is the bulk of it within the explorer, and CodeMirror in syntax-highlighter.

**Comment by erunion** [comment]
Upgraded this to Webpack 4.41.0 and we've got **considerably** smaller bundles.

```
Version: webpack 4.41.0
Time: 25626ms
Built at: 10/02/2019 1:25:24 PM
                  Asset      Size     Chunks                   Chunk Names
         demo-bundle.js  6.12 MiB       demo  [emitted]        demo
     demo-bundle.js.map  5.92 MiB       demo  [emitted] [dev]  demo
    reference-bundle.js  5.96 MiB  reference  [emitted]        reference
reference-bundle.js.map  5.77 MiB  reference  [emitted] [dev]  reference
```

```
$ du -hs packages/*/dist
1.4M	packages/api-explorer/dist
 32K	packages/api-logs/dist
```

@domharrington @rafegoldberg Can yall take another look at this?

**Comment by erunion** [comment]
Local development workflow should still be the same.

**Pull request merged**